diff --git a/src/setup.py b/src/setup.py
index c74f743..f31ca3c 100755
--- a/src/setup.py
+++ b/src/setup.py
@@ -870,7 +870,6 @@ print("Discovered Python/Cython sources, time: %.2f seconds." % (time.time() - t
 class sage_install(install):
     def run(self):
         install.run(self)
-        self.install_kernel_spec()
         log.info('Cleaning up stale installed files....')
         t = time.time()
         self.clean_stale_files()
