diff --git a/src/sage/interfaces/singular.py b/src/sage/interfaces/singular.py
index 5c5ca52..e9d1639 100644
--- a/src/sage/interfaces/singular.py
+++ b/src/sage/interfaces/singular.py
@@ -2279,7 +2279,6 @@ def generate_docstring_dictionary():
         sage: from sage.interfaces.singular import generate_docstring_dictionary
         sage: generate_docstring_dictionary()
     """
-    from sage.env import SAGE_LOCAL
 
     global nodes
     global node_names
@@ -2287,7 +2286,8 @@ def generate_docstring_dictionary():
     nodes.clear()
     node_names.clear()
 
-    singular_docdir = SAGE_LOCAL+"/share/info/"
+    import os
+    singular_docdir = os.environ['SINGULARPATH']+"/../share/info/"
 
     new_node = re.compile("File: singular\.hlp,  Node: ([^,]*),.*")
     new_lookup = re.compile("\* ([^:]*):*([^.]*)\..*")
