diff --git a/spkg-install b/spkg-install
index aa704c3..eef8412 100755
--- a/spkg-install
+++ b/spkg-install
@@ -79,12 +79,7 @@ if [[ -f "$DOT_SAGE/maxima_commandlist_cache.sobj" ]]; then
 fi
 
 
-# Install Maxima into ECL's library directory:
-ECLLIB=`ecl -eval "(princ (SI:GET-LIBRARY-PATHNAME))" -eval "(quit)"`
-echo
-echo "Now installing the Maxima library as '$ECLLIB/maxima.fas'..."
-cp -f "$CUR/src/src/binary-ecl/maxima.fas" "$ECLLIB/maxima.fas"
-check_error "Failed to install '$CUR/src/src/binary-ecl/maxima.fas' as '$ECLLIB/maxima.fas'."
+cp "$CUR/src/src/binary-ecl/maxima.fas" "$out/lib/maxima/$version/binary-ecl/"
 
 
 if [[ "$SAGE_SPKG_INSTALL_DOCS" = yes ]] ; then
