diff --git a/src/sage/all.py b/src/sage/all.py
index 2ea52ba..ca309ef 100644
--- a/src/sage/all.py
+++ b/src/sage/all.py
@@ -292,6 +292,8 @@ warnings.filterwarnings('ignore',
     module='.*(IPython|ipykernel|jupyter_client|jupyter_core|nbformat|notebook|ipywidgets|storemagic)')
 # Ignore warnings due to matplotlib-1.5 together with numpy-1.13
 warnings.filterwarnings('ignore', module='matplotlib[.]contour|numpy[.]ma[.]core')
+# Ignore matplotlib cache warnings (messing with doctests, not really relevant)
+warnings.filterwarnings('ignore', module='matplotlib[.]font_manager')
 # However, be sure to keep OUR deprecation warnings
 warnings.filterwarnings('default',
     '[\s\S]*See http://trac.sagemath.org/[0-9]* for details.')
