diff --git a/src/doc/common/conf.py b/src/doc/common/conf.py
index 2e115fa..1a73fe9 100644
--- a/src/doc/common/conf.py
+++ b/src/doc/common/conf.py
@@ -620,9 +620,9 @@ def call_intersphinx(app, env, node, contnode):
 
         sage: from sage.env import SAGE_DOC
         sage: thematic_index = os.path.join(SAGE_DOC, "html", "en", "thematic_tutorials", "index.html")
-        sage: for line in open(thematic_index).readlines():
-        ....:     if "padics" in line:
-        ....:         sys.stdout.write(line)
+        sage: for line in open(thematic_index).readlines():  # optional - htmldoc
+        ....:     if "padics" in line:                       # optional - htmldoc
+        ....:         sys.stdout.write(line)                 # optional - htmldoc
         <li><a class="reference external" href="../reference/padics/sage/rings/padics/tutorial.html#sage-rings-padics-tutorial" title="(in Sage Reference Manual: p-Adics ...)"><span>Introduction to the -adics</span></a></li>
     """
     debug_inf(app, "???? Trying intersphinx for %s"%node['reftarget'])
diff --git a/src/sage/misc/sagedoc.py b/src/sage/misc/sagedoc.py
index ebe1015..4ae8dd0 100644
--- a/src/sage/misc/sagedoc.py
+++ b/src/sage/misc/sagedoc.py
@@ -20,9 +20,9 @@ see :trac:`12849`::
 
     sage: from sage.env import SAGE_DOC
     sage: docfilename = os.path.join(SAGE_DOC, 'html', 'en', 'reference', 'calculus', 'sage', 'symbolic', 'expression.html')
-    sage: for line in open(docfilename):
-    ....:     if "#sage.symbolic.expression.Expression.numerical_approx" in line:
-    ....:         print(line)
+    sage: for line in open(docfilename):                                            # optional - htmldoc
+    ....:     if "#sage.symbolic.expression.Expression.numerical_approx" in line:   # optional - htmldoc
+    ....:         print(line)                                                       # optional - htmldoc
     <code class="descname">numerical_approx</code><span class="sig-paren">(</span><em>prec=None</em>, <em>digits=None</em>, <em>algorithm=None</em><span class="sig-paren">)</span>...
 
 Check that sphinx is not imported at Sage start-up::
@@ -791,7 +791,7 @@ def _search_src_or_doc(what, string, extra1='', extra2='', extra3='',
 
         sage: len(_search_src_or_doc('src', 'matrix\(', 'incidence_structures', 'self', 'combinat', interact=False).splitlines()) > 1
         True
-        sage: 'abvar/homology' in _search_src_or_doc('doc', 'homology', 'variety', interact=False)  # long time (4s on sage.math, 2012)
+        sage: 'abvar/homology' in _search_src_or_doc('doc', 'homology', 'variety', interact=False)  # optional - htmldoc, long time (4s on sage.math, 2012)
         True
         sage: 'divisors' in _search_src_or_doc('src', '^ *def prime', interact=False)
         True
@@ -1159,9 +1159,9 @@ def search_doc(string, extra1='', extra2='', extra3='', extra4='',
     counting the length of ``search_doc('tree',
     interact=False).splitlines()`` gives the number of matches. ::
 
-        sage: len(search_doc('tree', interact=False).splitlines()) > 4000  # long time
+        sage: len(search_doc('tree', interact=False).splitlines()) > 4000  # optional - htmldoc, long time
         True
-        sage: len(search_doc('tree', whole_word=True, interact=False).splitlines()) < 2000  # long time
+        sage: len(search_doc('tree', whole_word=True, interact=False).splitlines()) < 2000  # optional - htmldoc, long time
         True
     """
     return _search_src_or_doc('doc', string, extra1=extra1, extra2=extra2,
@@ -1331,7 +1331,7 @@ class _sage_doc:
 
     EXAMPLES::
 
-        sage: browse_sage_doc._open("reference", testing=True)[0]  # indirect doctest
+        sage: browse_sage_doc._open("reference", testing=True)[0]  # optional - htmldoc, indirect doctest
         'http://localhost:8000/doc/live/reference/index.html'
         sage: browse_sage_doc(identity_matrix, 'rst')[-107:-47]
         'Full MatrixSpace of 3 by 3 sparse matrices over Integer Ring'
@@ -1493,9 +1493,9 @@ class _sage_doc:
 
         EXAMPLES::
 
-            sage: browse_sage_doc._open("reference", testing=True)[0]
+            sage: browse_sage_doc._open("reference", testing=True)[0] # optional - htmldoc
             'http://localhost:8000/doc/live/reference/index.html'
-            sage: browse_sage_doc._open("tutorial", testing=True)[1]
+            sage: browse_sage_doc._open("tutorial", testing=True)[1]  # optional - htmldoc
             '.../html/en/tutorial/index.html'
         """
         url = self._base_url + os.path.join(name, "index.html")
